src/firstpass.rs

```diff
@@ -989,17 +1009,20 @@ impl<'a, 'b> FirstPass<'a, 'b> {
     /// Parse an ATX heading.
     ///
     /// Returns index of start of next line.
-    fn parse_atx_heading(&mut self, mut ix: usize, atx_level: HeadingLevel) -> usize {
+    fn parse_atx_heading(&mut self, start: usize, atx_level: HeadingLevel) -> usize {
+        let mut ix = start;
         let heading_ix = self.tree.append(Item {
-            start: ix,
-            end: 0, // set later
-            body: ItemBody::Heading(atx_level),
+            start,
+            end: 0,                    // set later
+            body: ItemBody::default(), // set later
         });
         ix += atx_level as usize;
         // next char is space or eol (guaranteed by scan_atx_heading)
         let bytes = self.text.as_bytes();
         if let Some(eol_bytes) = scan_eol(&bytes[ix..]) {
             self.tree[heading_ix].item.end = ix + eol_bytes;
+            self.tree[heading_ix].item.body =
```

Comment:
So here we allocating even when the header does not have any custom information, which could possibly be avoided if we wrap the `HeadingIndex` in an `Option`.